src/robot/libdocpkg/jsonbuilder.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright 2008-2015 Nokia Networks
+#  Copyright 2016-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+import os.path
+
+from robot.errors import DataError
+from robot.running.arguments import ArgumentSpec, ArgInfo
+
+from .model import LibraryDoc, KeywordDoc
+
+
+class JsonDocBuilder(object):
+
+    def build(self, path):
+        spec = self._parse_spec_json(path)
+        return self.build_from_dict(spec)
+
+    def build_from_dict(self, spec):
+        libdoc = LibraryDoc(name=spec['name'],
+                            doc=spec['doc'],
+                            version=spec['version'],
+                            type=spec['type'],
+                            scope=spec['scope'],
+                            doc_format=spec['doc_format'],
+                            source=spec['source'],
+                            lineno=int(spec.get('lineno', -1)))
+        libdoc.inits = self._create_keywords(spec['inits'])
+        libdoc.keywords = self._create_keywords(spec['keywords'])
+        return libdoc
+
+    def _parse_spec_json(self, path):
+        if not os.path.isfile(path):
+            raise DataError("Spec file '%s' does not exist." % path)
+        with open(path) as json_source:
+            libdoc_dict = json.load(json_source)
+        return libdoc_dict
+
+    def _create_keywords(self, keywords):
+        return [KeywordDoc(name=kw.get('name'),
+                           args=self._create_arguments(kw['args']),
+                           doc=kw['doc'],
+                           shortdoc=kw['shortdoc'],
+                           tags=kw['tags'],
+                           source=kw['source'],
+                           lineno=int(kw.get('lineno', -1))) for kw in keywords]
```

Comment:
I'd have just `_create_keyword()` and then call it within a list comprehension like

```python
libdoc.inits = [self._create_keyword(i) for i in spec['inits']]
libdoc.keywords = [self._create_keyword(k) for k in spec['keywords']]
```